packages/search-ui-workplace-search-connector/src/WorkplaceSearchAPIConnector.ts

```diff
@@ -177,24 +175,50 @@ class WorkplaceSearchAPIConnector {
     }
   }
 
-  onResultClick({
-    query,
-    documentId,
-    requestId,
-    tags = []
-  }: ResultClickParams): void {
-    tags = tags.concat("results");
-    return this.client.click({ query, documentId, requestId, tags });
+  onResultClick({ documentId, requestId }: ResultClickParams): void {
+    const apiUrl = `${this.enterpriseSearchBase}/api/ws/v1/analytics/event`;
+
+    this.performFetchRequest(apiUrl, {
+      type: "click",
+      document_id: documentId,
+      query_id: requestId,
+      content_source_id: null, // todo: where would this be available?
+      page: 1 // todo: where can i reach to meta?
+    });
   }
 
   onAutocompleteResultClick({
-    query,
     documentId,
-    requestId,
-    tags = []
+    requestId
   }: ResultClickParams): void {
-    tags = tags.concat("autocomplete");
-    return this.client.click({ query, documentId, requestId, tags });
+    const apiUrl = `${this.enterpriseSearchBase}/api/ws/v1/analytics/event`;
+
+    this.performFetchRequest(apiUrl, {
+      type: "click",
+      document_id: documentId,
+      query_id: requestId,
+      content_source_id: null, // where would this be available?
+      page: 1
+    });
+  }
+
+  async performFetchRequest(apiUrl: string, payload: any) {
```

Comment:
pulled up the main fetch call into a seperate method so all calls can benefit from the authorization handling.